Sources/SignalRClient/MessageBuffer.swift

```diff
@@ -10,11 +10,125 @@ actor MessageBuffer {
     private var totalMessageCount: Int = 0
     private var lastSendSequenceId: Int = 0
     private var nextSendIdx = 0
+    private var nextReceivingIdx: Int64 = 1
+    private var lastReceivedSequenceId: Int64 = 0
     private var dequeueContinuations: [CheckedContinuation<Bool, Never>] = []
     private var closed: Bool = false
+    private var reconnectInprogress: Bool = false;
+    private var waitForSequenceMessage: Bool = false;
 
-    init(bufferSize: Int) {
+    private var ackTimerHandle: DispatchWorkItem?
+
+    private var hubProtocol: HubProtocol;
+    private var connection: ConnectionProtocol;
+
+    init(bufferSize: Int, hubProtocol: HubProtocol, connection: ConnectionProtocol) {
         self.maxBufferSize = bufferSize
+        self.hubProtocol = hubProtocol
+        self.connection = connection
+    }
+
+    public func send(message: HubMessage) async throws -> Void {
+        let serializedMessage = try self.hubProtocol.writeMessage(message: message);
+
+        try await self.enqueue(content: serializedMessage);
+
+        if (!self.reconnectInprogress) {
+            do {
+                try await self.connection.send(serializedMessage);
```

Comment:
Is it possible that the message is enqueued, resent and sent again?

Reply:
this pr is replaced by https://github.com/dotnet/signalr-client-swift/pull/96/files
I think it won't. Before resent is done, `reconnectInprogress` is `true` so this `send` will be skipped